### ISSUE TITLE
tf3 c4 w1 neural style transfer: code implementation of losses

### ISSUE THREAD

**Issue by e7dud7e** [https-deeplearning-ai/tensorflow-3#20]
Before I make changes to the course 4 week 1 ungraded lab "neural style transfer", I wanted to do a sanity check, so I tagged @lmoroney , @cfav-dev  and @Mubsi .

The content loss is currently written as
```
  def get_content_loss(features, targets):
  	return tf.reduce_mean(tf.square(features - targets))
```

I think that the content loss should be 0.5 * reduce_sum and not reduce_mean
```
  def get_content_loss(features, targets):
  	return 0.5 * tf.reduce_sum(tf.square(features - targets))
```
If the reduce_mean is because it's working on a batch of items, then I think we should have a comment that explains why it's reduce_mean instead of reduce_sum.  For clarity either way, we should have the 0.5 scaling constant.

For style loss, I think the einstein notation should be written so that it does the matrix multiplication instead of an element-wise square.

It's currently written as
```
tf.linalg.einsum('bijc,bijd->bcd', input_tensor, input_tensor)
```
When I tested this on a small tensor
```
tmp = tf.constant([1,2,3,4], shape=[1,2,2,1])
```
It performed an element-wise square:
Notice I added the ->bijcd so that it wouldn't perform the reduce_sum, so I could verify what's going on before the sum.
```
tf.linalg.einsum('bijc,bijd->bijcd', tmp, tmp)
```
The result is an element-wise square
```
array([[[[[ 1]],
         [[ 4]]],
        [[[ 9]],
         [[16]]]]], dtype=int32)>
```

i think it should be:
```
tf.linalg.einsum('bijc,bjkd->bcd',tmp,tmp)
```
I tested this on the same [1,2,3,4] 2 by 2 matrix
I added the ->bjkcd to avoid reduce_sum, to see what's going on
```
tf.linalg.einsum('bijc,bjkd->bikcd',tmp,tmp)
```

The result is an A^T A matrix multiplication, which is what the gram matrix in theory should be:
```
array([[[[[ 7]],
         [[10]]],
        [[[15]],
         [[22]]]]]>
```
I verified this using matmul and transpose

```
tf.matmul(tf.transpose(tmp[0]),tmp[0])
```
It also gets the same matrix: 
```
array([[[[[ 7]],
         [[10]]],
        [[[15]],
         [[22]]]]]>
```
